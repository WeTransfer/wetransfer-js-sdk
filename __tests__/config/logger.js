const logger = require('../../src/config/logger');

describe('Logger configuration', () => {
  let log;
  beforeEach(() => {
    logger.setLoggerLevel('info');
    log = jest.spyOn(console._stdout, 'write').mockImplementation(() => {
      /* don't console.log anything */
    });
  });

  afterEach(() => {
    log.mockRestore();
  });

  describe('when silly level is set', () => {
    it('should log silly logs', () => {
      logger.setLoggerLevel('silly');
      logger.silly('This is a silly log');
      expect(log.mock.calls[log.mock.calls.length - 1]).toMatchSnapshot();
    });
  });

  describe('when error level is set', () => {
    it('should NOT log silly logs', () => {
      logger.setLoggerLevel('error');
      logger.silly('This is a silly log');
      expect(log).not.toHaveBeenCalled();
    });
  });

  describe('when setting a level that doesn\'t exist', () => {
    it('should keep default info level', () => {
      logger.setLoggerLevel('yolo');
      expect(log.mock.calls[log.mock.calls.length - 1]).toMatchSnapshot();
      expect(logger.transports[0].level).toBe('info');
    });
  });

  describe('when setting an empty level', () => {
    it('should keep default info level', () => {
      expect(logger.transports[0].level).toBe('info');
      logger.setLoggerLevel();
      expect(log.mock.calls[log.mock.calls.length - 1]).toMatchSnapshot();
      expect(logger.transports[0].level).toBe('info');
    });
  });
});
