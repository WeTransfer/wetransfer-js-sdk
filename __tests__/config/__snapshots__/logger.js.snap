// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logger configuration when setting a level that doesn't exist should keep default info level 1`] = `
Array [
  "[WeTransfer SDK] [33m\\"yolo\\" is not a valid logger level. Please specify one of the following levels: [[39m
[33m      'error',[39m
[33m      'warning',[39m
[33m      'info',[39m
[33m      'verbose,'[39m
[33m      'debug',[39m
[33m      'silly'[39m
[33m    ][39m
",
]
`;

exports[`Logger configuration when setting an empty level should keep default info level 1`] = `undefined`;

exports[`Logger configuration when silly level is set should log silly logs 1`] = `
Array [
  "[WeTransfer SDK] [35mThis is a silly log[39m
",
]
`;
